channels/OFF-CHAIN.md

```diff
@@ -1009,3 +1010,56 @@ contract update](../serializations.md#channel-off-chain-update-call-contract). I
 information needed for a contract call, including the contract address. The other participant co-signs
 the changes and the contract call is considered to be executed. Its results can be extracted from the calls tree in the state tree.
 
+### Contracts referring to on-chain data
+
+Contracts being used in channels off-chain calls have the exact same semantics
+as one being used on-chain. Because of the different environment though, those
+might have different results as the on-chain ones.
+
+Since there is no single source of truth, each participant considers their
+current view of the chain to be the correct one. This is essential for the
+trustless environment. Every off-chain contract call is based on the top of
+the chain, as it is seen by each participant. This could cause some differences
+in local contracts executions. If those can not be resolved, participants can
+always rely on the blockchain as an arbiter by using forcing of progress. Then
+the top is being used as it is seen by the Bitcoin-NG leader.
+
+It is worth mentioning that both local contracts' executions and the forced
+progress ones must be fully deterministic and this implies some restrictions
+on using on-chain objects in the off-chain contracts. This is especially true
+for the `Chain` API:
+
+* `Chain.coinbase` - the beneficiary of the latest block (either key or micro
+  block) as seen by the channel participant executing the contract call
+* `Chain.timestamp` - the time as in the latest block (either key or micro
+  block) as seen by the channel participant executing the contract call
```

Comment:
I see current doc of this kind of operations is defined:
* Coarsely in [Sophia primitives](https://github.com/aeternity/protocol/blob/25fde43eb9e958f677850e4c54228f71ebf16e16/contracts/sophia.md#contract-primitives)
* In a detailed manner in [VM doc](https://github.com/aeternity/protocol/blob/25fde43eb9e958f677850e4c54228f71ebf16e16/contracts/aevm.md#chain-specific-instructions)

It looks like this list of items (Chain.coinbase, ..., Chain.difficulty) may become another documentation point to keep consistency of. May I suggest to loosen the definition in this doc and detail the VM doc. Sample for this doc:
> This is especially true for the chain-related primitives (e.g. coinbase, timestamp, block height, difficulty). Please refer to the documentation of the applicable VM version.